libkbfs/journal_server.go

```diff
@@ -505,6 +506,102 @@ func (j *JournalServer) JournalStatus(tlfID TlfID) (TLFJournalStatus, error) {
 	return tlfJournal.getJournalStatus()
 }
 
+var errJournalStatusRetry = errors.New("Retry journal status")
+
+func (j *JournalServer) getUnflushedPaths(ctx context.Context, tlfID TlfID,
+	cpp chainsPathPopulator, branchStr string, start MetadataRevision,
+	end MetadataRevision) ([]string, error) {
+	// Get the range of revisions.  Note that since we're not doing
+	// this under lock, the journal could enter conflict mode, and we
+	// could end up fetching merged revisions from some other device.
+	// So we need to check that all the revisions were written by this
+	// device.  If not, retry.
+	var unflushedPaths []string
+	if end != MetadataRevisionUninitialized {
+		bid, err := ParseBranchID(branchStr)
+		if err != nil {
+			return nil, err
+		}
+
+		mStatus := Merged
+		if bid != NullBranchID {
+			mStatus = Unmerged
+		}
+		irmds, err := getMDRange(ctx, j.config, tlfID, bid, start, end, mStatus)
+		if err != nil {
+			return nil, err
+		}
+
+		// Make sure that all of them were signed by this device.
+		key, err := j.config.KBPKI().GetCurrentVerifyingKey(ctx)
```

Comment:
`tlfJournal` has a `key` field -- can you plumb that through instead of calling this? Just to make sure that we use the same key consistently